File: leetcode/Python/find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        sol, t = list(), 0
        for i in range(len(nums)):
            t = abs(nums[i])-1
            if nums[t] > 0:
                nums[t] *= -1
        for i in range(len(nums)):
            if nums[i]>0:
                sol.append(i+1)
        return sol
        

File: leetcode/C++/find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        vector<int> sol;
        for(int i=0,t=0; i<nums.size(); ++i){
            t=abs(nums[i])-1;
            if(nums[t]>0) nums[t]*=-1;
        }
        for(int i=0; i<nums.size(); ++i) if(nums[i]>0) sol.push_back(i+1);
        return sol;
    }
};